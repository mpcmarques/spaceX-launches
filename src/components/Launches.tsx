import styled from "@emotion/styled";
import React from "react";
import Launch from "../containers/Launch";
import { Launch as LaunchType } from "../types";

const LaunchesContainer = styled.section`
  .launches {
    height: 430px;
    overflow-y: scroll;

    div:last-of-type {
      margin-bottom: 0;
    }
  }
`;

const Launches: React.FC<{
  title: string;
  launches: Array<LaunchType>;
  loading: boolean;
}> = ({ launches, title, loading }) => {
  return (
    <LaunchesContainer>
      <h3>{title}</h3>
      {loading ? (
        <h4>Loading...</h4>
      ) : (
        <div className={"launches"}>
          {launches.length > 0
            ? launches.map((launch, index) => (
                <Launch key={index} launch={launch} />
              ))
            : "No launches"}
        </div>
      )}
    </LaunchesContainer>
  );
};

export default Launches;
