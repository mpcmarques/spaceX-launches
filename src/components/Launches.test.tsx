import React from "react";
import Launches from "./Launches";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("<Launches />", () => {
  it("renders without crashing given the required props", () => {
    const props = {
      title: "Upcoming",
      launches: [
        {
          success: false,
          name: "FalconSat",
          date_utc: "2006-03-24T22:30:00.000Z",
          upcoming: false,
          id: "5eb87cd9ffd86e000604b32a",
          rocket: {
            name: "Starship",
            flickr_images: ["img"],
          },
        },
      ],
      loading: false,
    };
    const wrapper = shallow(<Launches {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("shows no launches when it is loading", () => {
    const props = {
      title: "Upcoming",
      launches: [],
      loading: true,
    };
    const wrapper = shallow(<Launches {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("shows no launches when it doesnt fetch launches", () => {
    const props = {
      title: "Upcoming",
      launches: [],
      loading: false,
    };
    const wrapper = shallow(<Launches {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
