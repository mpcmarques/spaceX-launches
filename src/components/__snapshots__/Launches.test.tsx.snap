// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Launches /> renders without crashing given the required props 1`] = `
<Styled(section)>
  <h3>
    Upcoming
  </h3>
  <div
    className="launches"
  >
    <Launch
      key="0"
      launch={
        Object {
          "date_utc": "2006-03-24T22:30:00.000Z",
          "id": "5eb87cd9ffd86e000604b32a",
          "name": "FalconSat",
          "rocket": Object {
            "flickr_images": Array [
              "img",
            ],
            "name": "Starship",
          },
          "success": false,
          "upcoming": false,
        }
      }
    />
  </div>
</Styled(section)>
`;

exports[`<Launches /> shows no launches when it doesnt fetch launches 1`] = `
<Styled(section)>
  <h3>
    Upcoming
  </h3>
  <div
    className="launches"
  >
    No launches
  </div>
</Styled(section)>
`;

exports[`<Launches /> shows no launches when it is loading 1`] = `
<Styled(section)>
  <h3>
    Upcoming
  </h3>
  <h4>
    Loading...
  </h4>
</Styled(section)>
`;
